### PR TITLE
Faucet page

### PR DESCRIPTION
**Pull request by usulix**
##### Checklist

- [x] linter status: 100% pass
- [x] changes don't break existing behavior
- [x] commit message follows [commit guidelines]- add option to skip credentials on git search
- add faucet app and faucet_amount to config
- add faucet routes
- add faucet migrations
- add faucet templates
- add faucet model
- add faucet view and admin functions
- add faucet css additions
- add faucet specific js
- hook into existing function for faucet w3 errors
- add image for faucet page
- process faucet request

##### Refers/Fixes
https://github.com/gitcoinco/web/issues/184

### CODE REVIEW

**Comment by usulix** [comment]
@owocki 
I still get the BadRequest error on sending the email.
I also tried to send an email from the referenced whitepaper process and got a different error in the send_email function from mail.py, so I think that must be something in my config.

I do not get any token request or errors.. I can only imagine that would be CSRF token... but I did not see that anywhere.

testing boils down to visit localhost:8000/faucet and you should be able to create a request...
and visiting http://localhost:8000/_administration should allow you to click into faucet management and click process_me to pop up the meta_mask dialogue to okay the transfer

**Comment by owocki** [comment]
this still doesnt use the design provided by @pixelantdesign https://github.com/gitcoinco/web/issues/173#issuecomment-354347567

File: app/faucet/views.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    }
+
+    print request.POST.get('destinationAccount')
+    print request.POST
+    if obj.fulfilled:
+        return redirect('/_administrationfaucet/faucetrequest/')
+
+    if request.POST.get('destinationAccount', False):
+        obj.fulfilled = True
+        obj.save()
```
Comment (owocki) [inline app/faucet/views.py]: per the spec at https://github.com/gitcoinco/web/issues/184

> upon success, mark the faucet object as fulfilled and send an email notifying the recipient that their faucet request has been funded.

please make sure you trigger the email to the end user!

**Comment by usulix** [comment]
@owocki ,

re #173 (comment) -- since it was only mentioned under resources and not called out in the description, I was unaware that the mock ups existed.

These call for a slightly different workflow from what I have, but probably a desirable fix as it will require a user to actually sign  into their github - thus verifying that they are connected to the github account they are referencing.

Do we have psd's or similar of the design comps ?

**Comment by owocki** [comment]
> These call for a slightly different workflow from what I have, but probably a desirable fix as it will require a user to actually sign into their github - thus verifying that they are connected to the github account they are referencing.

yes, i agree this is important.. it'll prevent sybill attacks

> Do we have psd's or similar of the design comps ?

If you check https://github.com/gitcoinco/web/issues/173#issuecomment-354969169 @pixelantdesign put a ZIP file in the comments with any assets you'll need for this.

**Comment by owocki** [comment]
per @usulix message on slack.. he is super busy at work and needs some help getting this across the finish line..  i need to confirm with him, but i think yall should split the ETH.  anyone want?  @ethikz @thelostone-mc ?

**Comment by ethikz** [comment]
@owocki I'm looking into #184.  I believe these are both tied together but this one is pretty much finished based on comments on the referenced issue, am I correct in assuming that?

**Comment by owocki** [comment]
just moved over to this PR => https://github.com/gitcoinco/web/pull/407/

**Pull request closed without merging**